scripts/build_windows_shared.bat

```diff
@@ -0,0 +1,147 @@
+
+rem Remove to original folder after script is finished
```

Comment:
Why do we need an additional script? There is already one in `tools\build_pytorch_libs.bat`.

Reply:
The goal of the new script is to provide convinient end-user tool, that being clicked builds you Caffe2 in Windows with VIsual Studios. It doesn't mean replacing something existing for well-qualified users.

At the moment it supports only Visual Studio 2015.
However it is probably already more convenient for end-user with VS2015, then the existing two scripts
tools\build_pytorch_libs.bat
and
scripts\build_windows.bat.

The first script requires some user experience to be used correctly. In case of the proposed script both Debug and Release builds are automatically performed with correct CMAKE_C_COMPILER, CMAKE_CXX_COMPILER, CMAKE_LINKER, vcvars64.bat. On my PC it is required to have enough heap space to build caffe2_gpu.dll with pdbs (in Debug with all symbols). It is not that trivial for a non-experienced in Windows builds person to figure out how to do it.

The second script allows too few functionality to be built, although it is automatic as the proposed one.

The goal here is not to replace something existing, but to provide a tool for end-user to be simple to use (just click twice with your mouse) and allow building Caffe2 with maximum supported options.

The next step will be adding Visual Studio 2017 support, and the final is to add support of
glog, gflags, opencv, lmdb and probably leveldb (the last one is not that simple in Windows).

Reply:
However, if there is no need in such thing I'll of course remove it.

Reply:
As for the user-friendly script, I'm afraid it's not welcomed here because the platform-specific script will be eliminated in the long run. However, I've already written one [here](https://github.com/peterjc123/pytorch-scripts) for PyTorch. I'll grateful if you could contribute there.

Reply:
Fixed it, the file is deleted.
Thanks for the suggested repository. Sure, I'll contribute the script there.